vault/core_metrics.go

```diff
@@ -27,10 +27,13 @@ func (c *Core) metricsLoop(stopCh chan struct{}) {
 		select {
 		case <-emitTimer:
 			c.metricsMutex.Lock()
-			if c.expiration != nil {
+
+			// Emit on active node only
+			if c.expiration != nil && !c.perfStandby {
```

Comment:
Suggest `c.PerfStandby()` instead for locking.

Reply:
I think that some care is needed because c.stateLock can be held while acquiring c.metricsLock, so we shouldn't acquire them in the reverse order here.